app/services/api/concerns/declarations/shared_action.rb

```diff
@@ -0,0 +1,37 @@
+module API::Concerns::Declarations
+  module SharedAction
+    extend ActiveSupport::Concern
+
+    included do
+      include ActiveModel::Model
+      include ActiveModel::Attributes
+
+      attribute :lead_provider_id
+      attribute :declaration_id
+
+      validates :lead_provider_id, presence: { message: "Enter a '#/lead_provider_id'." }
+      validate :lead_provider_exists
+    end
+
+  private
+
+    def lead_provider
+      @lead_provider ||= LeadProvider.find_by(id: lead_provider_id) if lead_provider_id
+    end
+
+    def declaration
+      @declaration ||= Declaration.find_by!(api_id: declaration_id)
```

Comment:
We need to check if the lead provider owns the declaration.

The old ECF1 service we would only pass in `declaration` to the service if the lead provider owns the declaration:
https://github.com/DFE-Digital/early-careers-framework/blob/main/app/controllers/api/v3/participant_declarations_controller.rb#L102

For RECT we either only pass in verified declaration (and maybe dont need to pass in lead provider) or we check in the service whether the lead provider owns the declaration.